src/app/tests/TestReadInteraction.cpp

```diff
@@ -1840,13 +1838,12 @@ void TestReadInteraction::TestSubscribeWildcard(nlTestSuite * apSuite, void * ap
             err = engine->GetReportingEngine().SetDirty(dirtyPath);
             NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
 
-            ctx.DrainAndServiceIO();
-
-            //
-            // Not sure why I had to add this, and didn't have cycles to figure out why.
-            // Tracked in Issue #17528.
-            //
-            ctx.DrainAndServiceIO();
+            int last;
+            do
+            {
+                last = delegate.mNumAttributeResponse;
+                ctx.DrainAndServiceIO();
```

Comment:
There shouldn't be more than one `DrainAndServiceIO` call needed here.  The fact that we're doing more than one indicates there's a bug somewhere that we need to fix...  Why is this losing the comment pointing to the tracking issue?